lib/unit-ruby/util/api_resource.rb

```diff
@@ -101,6 +101,38 @@ def self.belongs_to(resource_name, class_name: nil)
       end
     end
 
+    def self.has_many(resource_name, class_name: nil)
+      class_name ||= resource_name.to_s.camelize
+
+      define_method(resource_name) do
+        singular_resource_name = resource_name.to_s.singularize.to_sym
+
+        # if there is only one of the resources, Unit will return the
+        # singular name of that relationship in their response.
+        # For example, if a deposit account has one customer, `customer`
+        # will be returned from Unit as a resource, whereas if there
+        # are more than one customer, then customers will be returned from Unit
+
+        # the below will translate these singular relationship
+        # into a plural relationship - ie `customer` -> `customers`
+
+        if relationships.keys.include?(singular_resource_name)
+
+          relationship_id = relationships.dig(singular_resource_name, :data, :id)
+
+          return [] unless relationship_id
+
+          return [relationship_id].map { |id| Kernel.const_get(class_name).find(id) }
+        else
+          relationship_ids = relationships.dig(resource_name.to_sym, :data).map { |x| x[:id] }
+
+          return [] if relationship_ids.empty?
+
+          return [relationship_ids].map { |id| Kernel.const_get(class_name).find(id) }
+        end
+      end
```

Comment:
I think this adds a lot of complexity trying to get around Unit's design decision to allow a singular and plural relationship.

My suggestion would just be to keep this narrow, and allow apps that use this SDK to handle those cases when there are mix of singular/plural entities themselves.

```suggestion
        resource_class = Kernel.const_get(class_name)
        relationships.dig(resource_name.to_sym, :data).map do |data|
          resource_class.build_resource_from_json_api(id)
        end
      end
```

Reply:
@ianyamey apologies, I thought I had responded to this. I guess my question would be how is the user of this service supposed to determine whether or not a particular deposit account has one or many users without throwing an error. If we're only providing a mapping of what's sent from them to us, calling `customers` on a deposit account that only has one will result in an error since the `relationship` returned from unit will only have a `customer` property.